### DISCUSSION TITLE
error SMTP laposte.net

### DISCUSSION THREAD

**Discussion by nonofr91** [authelia/authelia#7492] [Q&A] [answered]
Hello everyone,
I went from version 4.37 to version 4.38 and I cannot pass the Notifier/SMTP control. I have respected the new syntaxes but it does not work? Thank you in advance for any help you can give me.
I would like to confirm that I have another 4.37 installation which works correctly with the same SMTP. I deduce that it does not come from SMTP.
here is the configuration.yml session:
 smtp:
 ## The address of the SMTP server to connect to in the address common syntax.
 address: 'smtp://laposte.net:587'

 ## The connection timeout in the duration common syntax.
 timeout: '5 seconds'

 ## The username used for SMTP authentication.
 username: 'XXX.YYY@laposte.net'

 ## The password used for SMTP authentication.
 ## Can also be set using a secret: https://www.authelia.com/c/secrets
 password: 'zzzzzzzz'

 ## The sender is used to is used for the MAIL FROM command and the FROM header.
 ## If this is not defined and the username is an email, we use the username as this value. This can either be just
 ## an email address or the RFC5322 'Name <email address>' format.
 sender: 'Authelia <XXX.YYY@laposte.net>'

 ## HELO/EHLO Identify. Some SMTP Servers may reject the default of localhost.
 identify: 'localhost'

 ## Subject configuration of the emails sent. {title} is replaced by the text from the notifier.
 subject: '[Authelia] {title}'

 ## This address is used during the startup check to verify the email configuration is correct.
 ## It's not important what it is except if your email server only allows local delivery.
 startup_check_address: 'XXX.YYY@laposte.net'

 ## By default we require some form of TLS. This disables this check though is not advised.
 disable_require_tls: false

 ## Disables sending HTML formatted emails.
 disable_html_emails: false

And here is the error in the log file:

time="2024-07-05T09:48:05Z" level=error msg="Error occurred running a startup check" caller="github.com/authelia/authelia/v4/internal/commands/root.go:128 doStartupChecks" error="failed to dial connection: dial tcp 160.92.158.210:587: i/o timeout" provider=notification stack="github.com/authelia/authelia/v4/internal/commands/root.go:128 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:88  (*CmdCtx).RootRunE\ngithub.com/spf13/cobra@v1.8.1/command.go:985                  (*Command).execute\ngithub.com/spf13/cobra@v1.8.1/command.go:1117                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.1/command.go:1041                 (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\nruntime/internal/atomic/types.go:194                          (*Uint32).Load\nruntime/asm_amd64.s:1695                                      goexit"

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

It seems like you're encountering an issue with the SMTP startup check after upgrading from version 4.37 to 4.38. The error message indicates a timeout when trying to connect to the SMTP server. Here are a few steps you can take to troubleshoot and potentially resolve the issue:

1. **Check Network Connectivity**: Ensure that the server running Authelia can reach the SMTP server at `laposte.net` on port `587`. You can use tools like `telnet` or `nc` to verify this:
   ```sh
   telnet laposte.net 587
   ```

2. **Firewall and Security Groups**: Verify that there are no firewall rules or security group settings blocking outbound connections to the SMTP server.

3. **DNS Resolution**: Ensure that the DNS resolution for `laposte.net` is working correctly on the server running Authelia. You can use `nslookup` or `dig` to check this:
   ```sh
   nslookup laposte.net
   ```

4. **Configuration Syntax**: Double-check the configuration syntax. Although it seems correct, ensure there are no hidden characters or formatting issues.

5. **Disable Startup Check Temporarily**: As a temporary measure, you can disable the startup check to see if Authelia can start without it:
   ```yaml
   disable_startup_check: true
   ```

6. **Logs and Debugging**: Increase the logging level to debug to get more detailed information about the issue:
   ```yaml
   log_level: debug
   ```

7. **Version-Specific Issues**: There might be specific issues or changes in version 4.38 that affect the SMTP configuration. Reviewing the release notes or any related issues on the Authelia GitHub repository might provide insights.

Here is an example of a similar issue reported and the configuration used:
```yaml
notifier:
  disable_startup_check: false
  smtp:
    address: 'smtp://email-smtp.eu-west-2.amazonaws.com:465'
    timeout: 5s
    username: 'REDACTED'
    sender: 'Authelia <authelia@redacted.com>'
    subject: '[Authelia] {title}'
```
In this case, setting `disable_startup_check` to `true` allowed Authelia to start successfully.

For more detailed information, you can refer to the official documentation:
- [Authelia SMTP Configuration](https://www.authelia.com/configuration/notifications/smtp)

If the issue persists, please provide more details or logs to further diagnose the problem.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment]
Give `address: 'submission://laposte.net:587'` a try, might also help to show the config you had before if you still have it.

Also the i/o timeout indicates some kind of network issue.

**Reply by nonofr91** [reply]
My installation is in docker with a SWAG container. I have another container which uses the same SMTP without problem. I conclude that there is no network problem on the host.
Here is the same session in 4.37:

notifier:
  ## You can disable the notifier startup check by setting this to true.
  disable_startup_check: false

  ##
  ## File System (Notification Provider)
  ##
  ## Important: Kubernetes (or HA) users must read https://www.authelia.com/t/statelessness
  ##
  # filesystem:
  #   filename: /config/notification.txt

  ##
  ## SMTP (Notification Provider)
  ##
  ## Use a SMTP server for sending notifications. Authelia uses the PLAIN or LOGIN methods to authenticate.
  ## [Security] By default Authelia will:
  ##   - force all SMTP connections over TLS including unauthenticated connections
  ##      - use the disable_require_tls boolean value to disable this requirement
  ##        (only works for unauthenticated connections)
  ##   - validate the SMTP server x509 certificate during the TLS handshake against the hosts trusted certificates
  ##     (configure in tls section)
  smtp:
    # The SMTP host to connect to.
    host: smtp.laposte.net

    # The port to connect to the SMTP host on.
    port: 587

    # The connection timeout.
    timeout: 5s

    # The username used for SMTP authentication.
    username: XXX.YYY@laposte.net

    # The password used for SMTP authentication.
    # Can also be set using a secret: https://www.authelia.com/c/secrets
    password: zzzzzzz

    # The sender is used to is used for the MAIL FROM command and the FROM header.
    # If this is not defined and the username is an email, we use the username as this value. This can either be just
    # an email address or the RFC5322 'Name <email address>' format.
    sender: "Authelia <XXX.YYY@laposte.net>"

    # HELO/EHLO Identifier. Some SMTP Servers may reject the default of localhost.
    identifier: localhost

    # Subject configuration of the emails sent. {title} is replaced by the text from the notifier.
    subject: "[Authelia] {title}"

    # This address is used during the startup check to verify the email configuration is correct.
    # It's not important what it is except if your email server only allows local delivery.
    startup_check_address: XXX.YYY@laposte.net

    # By default we require some form of TLS. This disables this check though is not advised.
    disable_require_tls: false

    # Disables sending HTML formatted emails.
    disable_html_emails: false

**Reply by james-d-elliott** [reply]
Concluding that so early would be unwise. Please show the output of `telnet laposte.net 587` (the bot actually suggested this too, how neat)from the host the services are running on. I get connection timeouts (like authelia did) when trying that one:

```
~ telnet laposte.net 587
Trying 160.92.158.210...
telnet: Unable to connect to remote host: Connection timed out
```

**Reply by nonofr91** [reply]
I also get errors:
220 mlpnf0109.laposte.net ESMTP **************************
421 4.4.2 mlpnf0109.laposte.net Error: timeout exceeded


Loss of connection to host.

I did the test using local connection and 5G connection on my PC.
It seems that the problem is with laposte.net. I'll do some research because I can still send emails. Do they block pings and others on their servers?

**Reply by james-d-elliott** [reply]
Looks like it might be `smtp.laposte.net` actually, you need to change your config, try `address: 'submission://smtp.laposte.net:587'`

```
~ telnet smtp.laposte.net 587
Trying 160.92.124.65...
Connected to smtp.laposte.net.
Escape character is '^]'.
220 mlpnf0117.laposte.net ESMTP **************************
quit
221 2.0.0 Bye
Connection closed by foreign host.

**Reply by nonofr91** [reply, marked as answer]
thank you very much, that was it:
```
smtp:
    ## The address of the SMTP server to connect to in the address common syntax.
    address: 'smtp://smtp.laposte.net:587'``
```
